hack/update-codecgen.sh

```diff
@@ -143,7 +143,7 @@ for current in "${index[@]}"; do
   ###echo "DBG: running ${CODECGEN} -d 1234 -o ${base_generated_file} ${base_file}"
   ${CODECGEN} -d 1234 -o "${base_generated_file}" "${base_file}"
   # Add boilerplate at the beginning of the generated file.
-  sed 's/YEAR/2015/' "${initial_dir}/hack/boilerplate/boilerplate.go.txt" > "${base_generated_file}.tmp"
+  sed 's/YEAR/2016/' "${initial_dir}/hack/boilerplate/boilerplate.go.txt" > "${base_generated_file}.tmp"
```

Comment:
maybe we make a script that automatically generates a PR to change everything to 2017 next year? :)

but in all seriousness, you bring up a good point. this LG as-is. maybe #8830 will be fixed before next January?